component-api/src/main/java/org/talend/sdk/component/api/exception/ComponentException.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright (C) 2006-2020 Talend Inc. - www.talend.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.talend.sdk.component.api.exception;
+
+import lombok.Getter;
+
+public class ComponentException extends RuntimeException {
+
+    public static enum ErrorOrigin {
+        USER, // Error caused by user misconfiguration
+        BACKEND, // Error caused by backend
```

Comment:
May prefere either:
- EXECUTION
- RUNTIME
- RUNTIMEEXEC
-...
more than BACKEND.

Reply:
I use backend to tell the connector could not reach the backend it tries to access to. Just to tell, it's not connector's bad if it failed.